### PR TITLE
Fix DefaultChannelId MAC address parsing bug

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
DefaultChannelId provides a regular expression which validates if a user provided MAC address is valid. This regular expression may allow invalid MAC addresses and also not allow valid MAC addresses.

Modifications:
- Introduce a MacAddressUtil#parseMac method which can parse and validate the MAC address at the same time. The regular expression check before hand is additional overhead if we have to parse the MAC address.

Result:
Fixes https://github.com/netty/netty/issues/6132.

### CODE REVIEW

**Comment by Scottmitch** [comment]
@jo-kin - Please verify this fixes the issue.

**Comment by normanmaurer** [comment]
@Scottmitch there must be something wrong... Please check:

```
io.netty.channel.ChannelException: Unable to create Channel from class class io.netty.channel.local.LocalServerChannel
    at io.netty.bootstrap.BootstrapTest.testAsyncResolutionFailure(BootstrapTest.java:263)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class io.netty.channel.DefaultChannelId
    at io.netty.bootstrap.BootstrapTest.testAsyncResolutionFailure(BootstrapTest.java:263)
```

**Comment by Scottmitch** [comment]
@normanmaurer - issue should be fixed now.

File: common/src/main/java/io/netty/util/internal/MacAddressUtil.java [hunk 1]
```diff
@@ -110,27 +109,90 @@
...
+                bestMacAddr = Arrays.copyOf(bestMacAddr, EUI64_MAC_ADDRESS_LENGTH);
         }
 
         return bestMacAddr;
     }
 
     /**
+     * Returns the result of {@link #bestAvailableMac()} if non-{@code null} otherwise returns a random EUI-64 MAC
+     * address.
+     * @return the result of {@link #bestAvailableMac()} if non-{@code null} otherwise returns a random EUI-64 MAC
```
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/internal/MacAddressUtil.java]: @Scottmitch just remove the `@return` as it is the same as what is stated already ?

File: common/src/main/java/io/netty/util/internal/MacAddressUtil.java [hunk 2]
```diff
@@ -110,27 +109,90 @@
...
+        }
+
+        machineId[end] = (byte) Integer.parseInt(value.substring(j, value.length()), 16);
+
+        return machineId;
+    }
+
+    private static void validateMacSeparator(char separator) {
+        if (separator != ':' && separator != '-') {
+            throw new IllegalArgumentException("unsupported seperator: " + separator + " (expected: [:-])");
```
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/internal/MacAddressUtil.java]: @Scottmitch `:-]` looks nice and friendly ;)

**Comment by normanmaurer** [comment]
@Scottmitch just one comment.. otherwise LGTM

File: common/src/main/java/io/netty/util/internal/MacAddressUtil.java (hunk 1 above)
Comment (Scottmitch) [inline common/src/main/java/io/netty/util/internal/MacAddressUtil.java]: sure

**Comment by normanmaurer** [comment]
@Scottmitch ship it

**Comment by Scottmitch** [comment]
4.1 (3d11334)

**Pull request closed without merging**